src/backend/distributed/commands/utility_hook.c

```diff
@@ -706,10 +706,16 @@ ProcessUtilityInternal(PlannedStmt *pstmt,
 		if (IsA(parsetree, RenameStmt) && ((RenameStmt *) parsetree)->renameType ==
 			OBJECT_ROLE && EnableAlterRolePropagation)
 		{
-			ereport(NOTICE, (errmsg("not propagating ALTER ROLE ... RENAME TO commands "
-									"to worker nodes"),
-							 errhint("Connect to worker nodes directly to manually "
-									 "rename the role")));
+			if (EnablePropagationWarnings)
+			{
+				ereport(NOTICE, (errmsg(
+									 "not propagating ALTER ROLE ... RENAME TO commands "
+									 "to worker nodes"),
+								 errhint("Connect to worker nodes directly to manually "
+										 "rename the role")));
+			}
+
+			return;
```

Comment:
I don't think we are supposed to return here. Is that really needed?

Reply:
No, it is not, I removed it.